instrumentation/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -426,20 +450,27 @@ private SdkTracerProvider buildTracerProvider(SessionId sessionId, Application a
     private SpanExporter buildSpanExporter() {
         // TODO: Default to otlp...but how can we make endpoint and auth mandatory?
         SpanExporter defaultExporter = LoggingSpanExporter.create();
-        SpanExporter spanExporter = defaultExporter;
+        SpanExporter spanExporter = spanExporterCustomizer.apply(defaultExporter);
+
         DiskBufferingConfiguration diskBufferingConfiguration =
                 config.getDiskBufferingConfiguration();
         if (diskBufferingConfiguration.isEnabled()) {
             try {
-                spanExporter = createDiskExporter(defaultExporter, diskBufferingConfiguration);
+                SpanDiskExporter diskExporter =
+                        createDiskExporter(defaultExporter, diskBufferingConfiguration);
+                if (signalDiskExporterBuilder == null) {
+                    signalDiskExporterBuilder = SignalDiskExporter.builder();
+                }
+                signalDiskExporterBuilder.setSpanDiskExporter(diskExporter);
+                spanExporter = diskExporter;
```

Comment:
This seems overly complicated. It introduces a new field so that we can optionally create it in order to set the `diskExporter` here, which is really just a side effect that gets leveraged inside of the `setUpDiskBuffering()` method.

I think you should be able to get away with removing the new field and then factor out this chunk into a new method called `scheduleDiskReader()` or something, which does what the `setUpDiskBuffering()` method basically does. I think it will simplify some things, and this part will read more like a story:

* If the disk buffering is turned on
  * create the disk exporter
  * schedule the disk reader

Reply:
I see, the thing is that, by the time `setUpDiskBuffering` is called (now renamed to `scheduleDiskTelemetryReader`), the OpenTelemetry instance has already been created, so we'd lose the opportunity to set this disk exporter during the OTel initialization because it's immutable once it's all done (afaik), so the reason why I'm currently setting it up here is so that we can intercept the final, "normal" exporter before it's set into the span processor.

Maybe after [this PR](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1161) gets merged it won't need to be so complicated though. However, there's one more thing to be aware of regarding this part: `If the disk buffering is turned on` which is that even though the flag is enabled there's no guarantee that the buffering will actually work because there might be an issue when initializing the disk exporters (due to IO errors), causing the feature to get disabled even if it was requested to get enabled. I think @marandaneto mentioned something about this in a previous PR where we were pondering what would happen if some feature fails to get initialized during the RUM setup.

I guess a workaround that would allow to initialize the disk buffering exporters after the OpenTelemetry instance is created could be to create an exporter implementation that always wrapped the final exporter and would dynamically delegate to either the disk exporter or the "normal" exporter provided during the RUM setup depending on the config.